AEPAnalytics/Sources/Analytics.swift

```diff
@@ -310,86 +313,13 @@ public class Analytics: NSObject, Extension {
         }
     }
 
-    /// Stores the passed in visitor identifier in the analytics datastore via the `AnalyticsProperties`.
+    /// Dispatches event of type `EventType.analytics` and source `EventSource.responseContent` event with persisted ids and also updates analytics shared state.
     /// - Parameters:
-    ///     - event: The `Event` which triggered the visitor identifier update.
-    ///     - vid: The visitor identifier that was set.
-    private func updateVisitorIdentifier(event: Event, vid: String) {
-        if analyticsState.privacyStatus == .optedOut {
-            Log.debug(label: LOG_TAG, "updateVisitorIdentifier - Privacy is opted out, ignoring the update visitor identifier request.")
-            return
-        }
-
-        // persist the visitor identifier
-        analyticsProperties.setVisitorIdentifier(vid: vid)
-
-        // create a new analytics shared state and dispatch response for any extensions listening for AID/VID change
-        dispatchAnalyticsIdentityResponse(event: event)
-    }
-
-    /// Sends an analytics id request and processes the response from the server if there
-    /// is no currently stored AID. If an AID is already present in AnalyticsProperties,
-    /// the stored AID is dispatched and no network request is made.
-    /// - Parameters:
-    ///     - event: The `Event` which triggered the sending of the analytics id request.
-    private func retrieveAnalyticsId(event: Event) {
-        // check if analytics state contains an RSID and host OR if privacy opt-out. if so, update shared state with empty id.
-        if !analyticsState.isAnalyticsConfigured() || analyticsState.privacyStatus == .optedOut {
-            Log.debug(label: LOG_TAG, "sendAnalyticsIdRequest - Analytics is not configured or privacy is opted out, the analytics identifier request will not be sent.")
-            analyticsProperties.setAnalyticsIdentifier(aid: nil)
-            analyticsProperties.setVisitorIdentifier(vid: nil)
-            // create nil shared state  and dispatch this data in a response event for any extensions listening for AID/VID change
-            dispatchAnalyticsIdentityResponse(event: event)
-            return
-        }
```

Comment:
In C++ and Android Analytics we kept this section. However, I'm okay with removing it as it's an odd side-effect of calling this method. Let's add a task to remove this from C++ and Android as well so the behavior is the same.

Reply:
We already clear the IDs when privacy status changes to `optedOut`. This logic is redundant. I will add a task to remove this logic from C++ and Android.